docs/cip/CIP-8_Matrix_Packages.md

```diff
@@ -0,0 +1,209 @@
+## Motivation
+
+We want to foster an ecosystem of "knowledge packages," such that users can stitch together RAG programs by combining these packages with their own data. We will call such packages, and their format, "matrix." Drop a matrix package into your RAG app - now it knows kung fu.
+
+Many steps are necessary to reach this overarching goal, but the history of traditional software packaging clearly indicates that simple tools for both producing and consuming packages are prerequisite to healthy ecosystems. The scope of this CIP is an initial step towards that end: define a knowledge package format, and make the information held in Chroma portable into that format. The completeness criteria for this proposal is to define a full round trip, chroma -> matrix -> chroma.
+
+Ideally, the matrix package format could eventually be shared by the many tools in the emergent LLM/AI stack. This proposal seeks to balance that Chroma-agnostic goal with the reality that working, useful software is usually prerequisite to bootstrapping an ecosystem.
+## Impact
+
+This is a novel user path for Chroma, and as such does not affect any existing code or user behaviors. It may eventually affect workflows for developing and deploying Chroma-based apps, possibly even those that have no external dependencies.
+
+The main impact is on maintenance burden and (possible) compatibility guarantees. By creating what can be thought of as a portable serialization format for most of what is ordinarily stored in Chroma, it is likely that some properties which previously could have been considered implementation details, now will be assumed by users to be part of Chroma's public API surface.
+
+TODO examples
+
+## Proposed Change
+
+This is a proposal in two parts. First, we describe a package format. Then, we outline new chroma client capabilities and corresponding CLI subcommands that can round trip to and from this package format.
+
+Note that this CIP explicitly omits any actual dependency management behavior. While certain design choices here are motivated by preparing for that problem, directly addressing it is a problem for a future CIP.
+### Matrix Package Format
+A matrix package is a filesystem format consisting of two files, co-occurring within the same directory. Each file serves a particular, well-defined role:
+
+- `matrix.package.yaml`: defines the package name, version, and other common metadata, and certain metadata about the dataset in the package. Future iterations will add dependency declarations - the same kind of information that appears in `package.json` or `requirements.txt`. **Required**.
+- `matrix.data.parquet`: contains the package's dataset. Each row closely mirrors Chroma db content, containing original document source, computed embeddings, and metadata. **Required**, but will likely become optional in the future (more below)
```

Comment:
just a thought - one thing to consider is that we currently use sqlite and those _are_ portable. But we won't be using that in the Dis Sys so not sure if if we want to bet on it. Would love to learn more about this choice just for sake of exhaustiveness

Reply:
Yeah, reading over these comments, it's jumping out to me that it would be really valuable to get precise about "portability" means in this context.

The part of the puzzle i feel decently clear on right now is likely divergences between matrix packages and chroma db contents _after_ we get into actual dependencies. We're treating that as out of scope for this proposal, but it's probably worth making a note about.